terraform/main.tf

```diff
@@ -0,0 +1,144 @@
+
+provider "aws" {
+  region = var.region
+  default_tags {
+    tags = var.default_tags_tf
+  }
+}
+
+
+resource "aws_kms_key" "this" {
+  description = "KMS key for MSK SCRAM credentials"
+  tags        = var.tags
+}
+
+resource "aws_kms_alias" "this" {
+  name          = "alias/${var.identifier}-msk-scram"
+  target_key_id = aws_kms_key.this.key_id
+}
+
+resource "aws_cloudwatch_log_group" "this" {
+  count = var.logging_info != null ? 1 : 0
+
+  name              = "${var.identifier}-msk-broker-logs"
+  retention_in_days = var.logging_info.broker_logs.cloudwatch_logs.retention_in_days
+  tags              = var.tags
+}
+
+resource "aws_msk_configuration" "this" {
+  name              = "${var.identifier}-${replace(var.kafka_version, ".", "-")}"
+  kafka_versions    = [var.kafka_version]
+  server_properties = var.server_properties
+
+  lifecycle {
+    create_before_destroy = true
+  }
+}
+
+resource "aws_secretsmanager_secret" "this" {
+  for_each   = var.scram_users
+  name       = "AmazonMSK_${var.identifier}-${each.key}"
+  tags       = var.tags
+  kms_key_id = aws_kms_key.this.key_id
+}
+
+resource "aws_secretsmanager_secret_version" "this" {
+  for_each  = var.scram_users
+  secret_id = aws_secretsmanager_secret.this[each.key].id
+  # enforce the order of the keys in the JSON object
+  secret_string = jsonencode({
+    for k in sort(keys(each.value)) : k => each.value[k]
+  })
```

Comment:
`jsonencode` should already iterate map keys lexicographically? Even reconstruct map with reversed insertion order, the result should be the same. Can you check other result of `jsonencode` like 

```hcl
policy = jsonencode({
    Version = "2012-10-17"
    Statement = [
```

see if `Version` is after `Statement` in the final string.

Reply:
jupp, you're right, `jsonencode` already sorts the keys. Changed in [96674c9](https://github.com/app-sre/er-aws-msk/pull/167/commits/96674c988ce27b2d1329be27821cbb898a2526b9)